diff --git a/hci_core_original.c b/hci_core_original.c
index a39ed39..883235d 100644
--- a/hci_core_original.c
+++ b/hci_core_original.c
@@ -149,7 +149,6 @@ static int hci_init1_req(struct hci_request *req, unsigned long opt)
 static void bredr_setup(struct hci_request *req)
 {
         __le16 param;
-        __u8 flt_type;
 
         /* Read Buffer Size (ACL mtu, max pkt, etc.) */
         hci_req_add(req, HCI_OP_READ_BUFFER_SIZE, 0, NULL);
@@ -170,10 +169,15 @@ static void bredr_setup(struct hci_request *req)
         hci_req_add(req, HCI_OP_READ_CURRENT_IAC_LAP, 0, NULL);
 
         /* Clear Event Filters */
-        flt_type = HCI_FLT_CLEAR_ALL;
-        hci_req_add(req, HCI_OP_SET_EVENT_FLT, 1, &flt_type);
   /* Clear Event Filters if the device is able to do so */
 
+        if (!test_bit(HCI_QUIRK_BROKEN_ERR_DATA_REPORTING, &req->hdev->quirks)) {
+                __u8 flt_type;
+
+                flt_type = HCI_FLT_CLEAR_ALL;
+                hci_req_add(req, HCI_OP_SET_EVENT_FLT, 1, &flt_type);
+        }
+
         /* Connection accept timeout ~20 secs */
         param = cpu_to_le16(0x7d00);
         hci_req_add(req, HCI_OP_WRITE_CA_TIMEOUT, 2, &param);
